### PR TITLE
Add some multivariate metrics and refactor old ones

### PR DESCRIPTION
**Pull request by MadHypnofrog**

### CODE REVIEW

File: ITMO_FS/filters/multivariate/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
 from .MultivariateFilter import MultivariateFilter
-from .measures import *
+from .measures import GLOB_MEASURE, MIM, MRMR, JMI, CIFE, MIFS, CMIM, ICAP, DCSF, CFR, MRI, IWFS
```
Comment (LastShekel) [inline ITMO_FS/filters/multivariate/__init__.py:2]: Why?

File: ITMO_FS/filters/multivariate/measures.py [hunk 2]
```diff
@@ -1,63 +1,87 @@
...
-		np.sum(np.apply_along_axis(calc_conditional_mutual_information, 0, X[:, selected_features], X[:, free_feature], y)))(free_features)
-	return relevance - np.maximum(redundancy - cond_dependancy, 0.) 
+	cond_dependency = np.vectorize(lambda free_feature : \
+		np.sum(np.apply_along_axis(conditional_mutual_information, 0, X[:, selected_features], X[:, free_feature], y)))(free_features)
+	return relevance - np.maximum(redundancy - cond_dependency, 0.) 
+
+def DCSF(selected_features, free_features, X, y):
+	if selected_features.size == 0:
+		return np.vectorize(lambda x : 0)(free_features)
+	vectorized_function = lambda free_feature : np.sum( \
```
Comment (LastShekel) [inline ITMO_FS/filters/multivariate/measures.py:44]: duplication

File: ITMO_FS/filters/multivariate/measures.py [hunk 3]
```diff
@@ -1,63 +1,87 @@
...
+		np.apply_along_axis(lambda z, x, y: conditional_mutual_information(x, y, z), 0, X[:, selected_features], X[:, free_feature], y) +
+		np.apply_along_axis(conditional_mutual_information, 0, X[:, selected_features], y, X[:, free_feature]) -
+		matrix_mutual_information(X[:, selected_features], X[:, free_feature]))
+	return np.vectorize(vectorized_function)(free_features)
+
+def CFR(selected_features, free_features, X, y):
+	if selected_features.size == 0:
+		return np.vectorize(lambda x : 0)(free_features)
+	vectorized_function = lambda free_feature : np.sum( \
```
Comment (LastShekel) [inline ITMO_FS/filters/multivariate/measures.py:53]: duplication

File: ITMO_FS/filters/multivariate/__init__.py (hunk 1 above)
Comment (MadHypnofrog) [reply ITMO_FS/filters/multivariate/__init__.py]: reverted back to *

File: ITMO_FS/filters/multivariate/measures.py (hunk 2 above)
Comment (MadHypnofrog) [reply ITMO_FS/filters/multivariate/measures.py:44]: discussed, there are no duplications

File: ITMO_FS/filters/multivariate/measures.py (hunk 3 above)
Comment (MadHypnofrog) [reply ITMO_FS/filters/multivariate/measures.py:53]: discussed, there are no duplications

**Pull request merged**